src/snowflake/snowpark/window.py

```diff
@@ -34,20 +34,29 @@
     from collections.abc import Iterable
 
 
-def _convert_boundary_to_expr(start: int, end: int) -> Tuple[Expression, Expression]:
-    if start == 0:
-        boundary_start = CurrentRow()
-    elif start <= Window.UNBOUNDED_PRECEDING:
-        boundary_start = UnboundedPreceding()
+def _convert_boundary_to_expr(
+    start: Union[int, "snowflake.snowpark.Column"],
+    end: Union[int, "snowflake.snowpark.Column"],
+) -> Tuple[Expression, Expression]:
+    if isinstance(start, int):
+        if start == 0:
+            boundary_start = CurrentRow()
+        elif start <= Window.UNBOUNDED_PRECEDING:
+            boundary_start = UnboundedPreceding()
+        else:
+            boundary_start = Literal(start)
     else:
```

Comment:
```suggestion
    elif isinstance(start, Column):
```
Might be useful to throw a descriptive error if the typing doesn't match expectation. Similar for end.

Reply:
good point, added